pkg/i2gw/providers/gce/gce_extensions.go

```diff
@@ -0,0 +1,188 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gce
+
+import (
+	"context"
+	"encoding/json"
+
+	gkegatewayv1 "github.com/GoogleCloudPlatform/gke-gateway-api/apis/networking/v1"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/ir"
+	"github.com/kubernetes-sigs/ingress2gateway/pkg/i2gw/notifications"
+	apiv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/ingress-gce/pkg/annotations"
```

Comment:
Very tiny nit: The `annotations` name keeps throwing me off because I think I'm reading the contents of an annotations field, and not some kind of constants, maybe a different name would help:
```suggestion
	gceAnnotations "k8s.io/ingress-gce/pkg/annotations"
```

Reply:
Actually I think we can skip this import, and just add imported variable as constance because we are only using `backendConfigKey`, `betaBackendConfigKey`, and the following struct
```
type backendConfigs struct {
	Default string
	Ports   map[string]string
}
```

I think it would be better if we just maintain them ourselves. Updated to include as constant in types.go and gce_extensions.go

Reply:
SGTM, I like that approach